components/micro-gateway-core/src/main/ballerina/src/gateway/filters/pre_authn_filter.bal

```diff
@@ -173,3 +175,22 @@ function getAuthenticationProviderType(string authHeader) returns (string) {
         return AUTH_SCHEME_OAUTH2;
     }
 }
+
+function preserveRequestHeaders(http:Request request) {
+    boolean preserveUserAgentHeader = getConfigBooleanValue(SERVER_PRESERVE_HEADERS_ID,
+                                                            SERVER_PRESERVE_HEADERS_USER_AGENT,
+                                                            DEFAULT_SERVER_PRESERVE_HEADERS_USER_AGENT);
+    if (preserveUserAgentHeader) {
+        request.setHeader("User-Agent", request.userAgent);
+    }
+}
+
+function preserveResponseHeaders(http:Response response) {
+    boolean preserveServerHeader = getConfigBooleanValue(SERVER_PRESERVE_HEADERS_ID,
```

Comment:
Lets read the value only once and store it and use it

Reply:
https://github.com/wso2/product-microgateway/pull/1577/commits/85e1f6f660e528931e77fb079ff35d3d221b3c1c